datalake_api/querier.py

```diff
@@ -196,7 +210,8 @@ def query_by_time(self, start, end, what, where=None, cursor=None):
             buckets = buckets[i:]
 
         for b in buckets:
-            kwargs = self._prepare_time_bucket_kwargs(b, what)
+            kwargs = self._prepare_time_bucket_kwargs(b, what,
+                                                      limit=MAX_RESULTS/2)
```

Comment:
So, MAX_RESULTS is not meant to be the max results that we fetch from dynamo. It's meant to be the max results we return to the user in a page. And how we pick up the search where we left on from page to page is managed by the cursor.

Here's what happens in the specific case that you describe: We do one query and get the first 50 results back from the first bucket, and dynamo gives us a LastEvaluatedKey, indicating that the bucket has more results. We'll go ahead and grab them because we have at least MAX_RESULTS/2 head room in our page. That'll give us the next 40 results and no LastEvaluatedKey. We'll stop there because we don't have the potential room for 50 more records, and we'll return a page of 90 results to the user. We encode the next bucket that we have to check in the cursor. Then, when the user passes us that cursor back, we'll fetch the 10 in the next bucket.